server/main.py

```diff
@@ -94,3 +121,28 @@ async def calculate_spectrum(payload: Payload):
                 "units": spectrum.units[payload.mode],
             },
         }
+
+#[Date] _[database] _[molecule]_[temperature]K_[pressure]atm
+@app.post("/download-spectrum")
+async def download_spec(payload: Payload):
+    print(payload)
+    date = datetime.datetime.now().strftime("%Y_%m_%d_%H_%M_%S");
+    file_name = f"downloaded_spectrum/{date}_{payload.database}_{payload.tgas}k_{payload.pressure}atm.spec"
+    # file_path=f"create_spectrum/{file_name}"
+    print(f".spec file is created here{file_name}")
+   
+    try:
+        spectrum = calculate_spectrum(payload)
+       
+        if payload.use_simulate_slit is True:
+            print("Applying simulate slit")
+            spectrum.apply_slit(payload.simulate_slit, "nm")
+    except radis.misc.warning.EmptyDatabaseError:
+        return {"error": "No line in the specified wavenumber range"}
+    except Exception as exc:
+        print("Error", exc)
+        return {"error": str(exc)}
+    else:
+        spectrum.store(file_name, compress=True, if_exists_then='replace')
+        return FileResponse(file_name, media_type='application/octet-stream', filename=file_name)
```

Comment:
Wouldn't we want to delete the file after we return it back to the user? I wouldn't want a bunch of files getting created over time that will use up the server's disk space.

Reply:
Maybe relevant: https://stackoverflow.com/a/64717120